src/lightning_fabric/wrappers.py

```diff
@@ -189,3 +190,17 @@ def _process_optimizer_zero_grad_kwargs(optimizer: Optimizer, kwargs: Dict[str,
         # Some optimizers out there, for example DeepSpeedZeroOptimizer, use a different name than PyTorch
         kwargs["set_grads_to_None"] = kwargs.pop("set_to_none")
     return kwargs
+
+
+def _unwrap_objects(collection: T_collection) -> T_collection:
```

Comment:
This type annotation isn't correct, as you aren't returning the same collection. I think mypy doesn't complain because typing gets lost through `apply_to_collection` and treats it as `Any`.

So maybe you should annotate it as `Any`, or leave it as it is.

Reply:
I knew I would get roasted on this haha.
Changed to Any